### PR TITLE
Updates initial shared state onRegistered (resolves #41)

### LINKED ISSUE #41: revisit bootup sequence
- share persisted data on shared state version 0
- revisit bootup sequence and actions on config updates

### PR DESCRIPTION
**Pull request by emdobrin**
## Description

- Decoupled the boot up from the config response event and Analytics shared state with previously cached values. Config response events will now post the updated shared state only on privacy optout (clearing the ids) - this is also in line with 1.x.
- Updated few unrelated tests that were assuming the first config event will share initial state

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/analytics/src/test/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsExtensionTests.kt [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        reset(mockApi)
+        reset(mockDataStore)
+        reset(mockDatabase)
+        reset(mockNamedCollection)
+        mockedStaticServiceProvider.close()
+        reset(mockServiceProvider)
+    }
+
+    @Test
+    fun `onRegistered() should call createSharedState for version 0`() {
```
Comment (kevinlind) [inline code/analytics/src/test/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsExtensionTests.kt:83]: Can you add another test to verify an empty shared state is created if there is no AID for VID values?  I know the code always creates the shared state now, but if the code is updated in the future it is good to have a test which asserts an empty state is created.

**Review by kevinlind — APPROVED** [review]
